server/message.go

```diff
@@ -0,0 +1,57 @@
+package server
+
+import (
+	"errors"
+
+	"github.com/coniks-sys/coniks-go/utils"
+)
+
+var (
+	ErrorMalformedClientMessage = errors.New("[coniks] Malformed client request")
+	ErrorNameExisted            = errors.New("[coniks] Registering identity is already registered")
+	ErrorInternalServer         = errors.New("[coniks] Internal server error")
+)
+
+const (
+	RegistrationType = "registration"
+)
+
+type Request struct {
+	Type    string
+	Request interface{}
+}
+
+type RegistrationRequest struct {
+	Username               string `json:"username"`
+	Key                    string `json:"key"`
+	KeyType                string `json:"key_type,omitempty"`
+	AllowUnsignedKeychange bool   `json:"allow_unsigned_key_change,omitempty"`
+	AllowPublicLookup      bool   `json:"allow_public_lookup,omitempty"`
+}
+
+type Response interface{}
+
+type ErrorResponse struct {
+	Error string
+}
+
+type RegistrationResponse struct {
+	TempraryBinding string `json:"tb"`
+}
+
+func NewErrorResponse(err error) *ErrorResponse {
+	return &ErrorResponse{Error: err.Error()}
+}
+
+func (reg *RegistrationRequest) serialize() []byte {
+	// serialize scheme:
```

Comment:
> Actually, now that I think about it some more, does the server need an interface to store things like RegistrationRequests in the DB? I'm not sure why the server should store these, so I don't know if messagekv.go would be needed. Do you think the server should store all the requests it receives from the client in the DB?

I think we do need a DB, not for registration request, but for users' data blob including users' policies. The server needs to store users' policies and also cannot store it in the tree, right?